src/v/cloud_topics/BUILD

```diff
@@ -108,13 +108,32 @@ redpanda_cc_library(
     ],
 )
 
+redpanda_cc_library(
+    name = "data_plane_api",
+    hdrs = [
+        "data_plane_api.h",
```

Comment:
We decided to update the naming here right and the whole app should not be the just the data plane API?